### PR TITLE
Bump dependencies in Go services

### PR DESCRIPTION
**Pull request by pellared**
### Background 

I noticed that the `pb.go` files where not regenerated for a long time and thus the generated code was using deprecated Go module (`github.com/golang/protobuf`)

I think that without regenerating it might not even work with `google.golang.org/grpc v1.66.0`. At least this is what I noticed when working on https://github.com/signalfx/microservices-demo/pull/152.

### Fixes 

N/A

### Change Summary

- Bump Go version in `go.mod` files to `1.23` (the `Dockerfile` are already using Go 1.23.
- Update the `genproto.sh` so that it works fine with latest protoc tooling. References:
  - https://grpc.io/docs/languages/go/quickstart/
  - https://protobuf.dev/reference/go/go-generated/

### Additional Notes

- nit: I also replaced `as` to `AS` in Dockerfiles to get rid of `docker build` warning

### CODE REVIEW

**Comment by pellared** [comment]
[header-check](https://github.com/GoogleCloudPlatform/microservices-demo/pull/2716/checks?check_run_id=29713601141) failed for generated files. I am not sure if I should do anything with it. Feel free to edit if necessary.

**Comment by pellared** [comment]
Should I revert the Go version to 1.22 or update to Go 1.23 in places that I have missed?

File: src/shippingservice/go.mod [hunk 1]
```diff
@@ -1,43 +1,40 @@
 module github.com/GoogleCloudPlatform/microservices-demo/src/shippingservice
 
-go 1.21
-
-toolchain go1.23.0
```
Comment (bourgeoisor) [inline src/shippingservice/go.mod:5]: Is the issue that you removed the toolchain line here, so it defaults to trying to find a toolchain matching the Go version on line 3 (`1.23`) which is invalid, maybe?: https://github.com/golang/go/issues/62278#issuecomment-1698829945

Comment (pellared) [reply src/shippingservice/go.mod:5]: No 😉 

This is caused by https://github.com/golang/go/issues/68971#issuecomment-2299062164

The GitHub workflows are using Go 1.22.2.

I will bump Go version in GitHub Actions.

Comment (pellared) [reply src/shippingservice/go.mod:5]: https://github.com/GoogleCloudPlatform/microservices-demo/pull/2716/commits/660990d0ac7d68680de4901074e46c5690a799b9

File: protos/demo.proto [hunk 2]
```diff
@@ -16,6 +16,8 @@ syntax = "proto3";
 
 package hipstershop;
 
+option go_package = "github.com/GoogleCloudPlatform/miscroservices-demo/hipstershop";
```
Comment (bourgeoisor) [inline protos/demo.proto:19]: typo?

Comment (pellared) [reply protos/demo.proto:19]: yes 😉

Comment (pellared) [reply protos/demo.proto:19]:
```suggestion
option go_package = "github.com/GoogleCloudPlatform/microservices-demo/hipstershop";
```

**Review by bourgeoisor — APPROVED** [review]
LGTM

**Review by mike-ensor — COMMENTED** [review]
LGTM (posthumously, should have refreshed before review)

**Pull request merged**